wagtail/admin/wagtail_hooks.py

```diff
@@ -399,7 +410,55 @@ def page_header_buttons(page, page_perms, next_url=None):
                 "aria-label": _("Add a child page to '%(title)s' ")
                 % {"title": page.get_admin_display_title()},
             },
-            priority=30,
+            priority=40,
+        )
+    if page_perms.can_delete():
+        url = reverse("wagtailadmin_pages:delete", args=[page.id])
+
+        # After deleting the page, it is impossible to redirect to it.
+        if next_url == reverse("wagtailadmin_explore", args=[page.id]):
+            next_url = None
+
+        if next_url:
+            url += "?" + urlencode({"next": next_url})
+
+        yield Button(
+            _("Delete"),
+            url,
+            icon_name="bin",
+            attrs={
+                "title": _("Delete page '%(title)s'")
+                % {"title": page.get_admin_display_title()}
+            },
+            priority=50,
+        )
+    if page_perms.can_unpublish():
+        url = reverse("wagtailadmin_pages:unpublish", args=[page.id])
+        if next_url:
+            url += "?" + urlencode({"next": next_url})
+
+        yield Button(
+            _("Unpublish"),
+            url,
+            icon_name="download-alt",
+            attrs={
+                "title": _("Unpublish page '%(title)s'")
+                % {"title": page.get_admin_display_title()}
+            },
+            priority=60,
+        )
+    if is_parent and page_perms.can_reorder_children():
```

Comment:
Something to revisit – I’ think we’d need to make sure `is_parent` is set appropriately between this and similar hooks, and potentially deprecate it in a few places.